sttp/ticks/Ticks.go

```diff
@@ -28,99 +30,43 @@ import (
+// Leap seconds are exposed, but are silently discarded upon conversion to Go or Unix timestamps.
 type Ticks uint64
 
-// Min is the minimum value for Ticks. It represents UTC time 01/01/0001 00:00:00.000.
+// Equivalent to UTC time 01/01/0001 00:00:00.000.
 const Min Ticks = 0
-
-// Max is the maximum value for Ticks. It represents UTC time 12/31/1999 11:59:59.999.
+// Max is the maximum value for Ticks. It represents UTC time 12/31/9999 11:59:59.999.
 const Max Ticks = 3155378975999999999
 
-// PerSecond is the number of Ticks that occur in a second.
-const PerSecond Ticks = 10000000
-
-// PerMillisecond is the number of Ticks that occur in a millisecond.
-const PerMillisecond Ticks = PerSecond / 1000
-
-// PerMicrosecond is the number of Ticks that occur in a microsecond.
-const PerMicrosecond Ticks = PerSecond / 1000000
-
-// PerMinute is the number of Ticks that occur in a minute.
-const PerMinute Ticks = 60 * PerSecond
-
-// PerHour is the number of Ticks that occur in an hour.
-const PerHour Ticks = 60 * PerMinute
+// Ticks are every 100ns == 0.1us
+const PerMicrosecond = 10
+const PerMillisecond Ticks = PerMicrosecond*1000
 
-// PerDay is the number of Ticks that occur in a day.
-const PerDay Ticks = 24 * PerHour
-
-// LeapSecondFlag is the flag (64th bit) that marks a Ticks value as a leap second, i.e., second 60 (one beyond normal second 59).
 const LeapSecondFlag Ticks = 1 << 63
-
-// LeapSecondDirection is the flag (63rd bit) that indicates if leap second is positive or negative; 0 for add, 1 for delete.
 const LeapSecondDirection Ticks = 1 << 62
-
-// ValueMask defines all bits (bits 1 to 62) that make up the value portion of a Ticks that represent time.
 const ValueMask Ticks = ^LeapSecondFlag & ^LeapSecondDirection
 
-// UnixBaseOffset is the Ticks representation of the Unix epoch timestamp starting at January 1, 1970.
+// Ticks representation of the Unix epoch, equivalent to FromUnixNs(0).
 const UnixBaseOffset Ticks = 621355968000000000
 
-// TimeFormat is the standard time.Time format used for a Ticks value.
 const TimeFormat string = "2006-01-02 15:04:05.999999999"
-
-// ShortTimeFormat is the standard time.Time format used for showing just the timestamp portion of a Ticks value.
 const ShortTimeFormat string = "15:04:05.999"
 
-// TimestampValue gets the timestamp portion of the Ticks value, i.e.,
-// the 62-bit time value excluding any leap second flags.
-func (ticks Ticks) TimestampValue() int64 {
-	return int64(ticks & ValueMask)
+func (t Ticks) TimestampValue() int64 {
+	return int64(t&ValueMask)
 }
 
-// ToTime converts a Ticks value to standard Go Time value.
-func ToTime(ticks Ticks) time.Time {
-	return time.Unix(0, int64((ticks-UnixBaseOffset)&ValueMask)*100).UTC()
+// Converts a unix nanoseconds timestamp into a Ticks value.
+func FromUnixNs(ns uint64) Ticks {
+	return Ticks(ns / 100) + UnixBaseOffset
```

Comment:
Instead of replacing the ToTime function, we should just add the FromUnixNs function.  That way we don't suggest breaking changes to a public API.

Also, I suggest naming `FromUnixNs` as `FromUnixNano` just to be consistent with `time.UnixNano()` method in the golang std lib.